File: backend/backend/src/main/java/Shapes/Circle.java
package Shapes;

import HelpingClasses.Sizes.CircleSize;
import HelpingClasses.Sizes.Size;

public class Circle extends ClosedShape
{
    public Circle(int id)
    {
        super(id);
    }

    @Override
    public void Clone(Shape shape)
    {
        super.Clone(shape);
    }

    @Override
    public void SetSize(Size size)
    {
        this.MySize = (CircleSize) size;
    }
    @Override
    public Size GetSize()
    {
        return this.MySize;
    }


}

File: frontend/paint_frontend/src/app/classes/Circle.ts
import { CircleSize } from "./CircleSize";
import { ClosedShape } from "./ClosedShape";
import { Position } from "./Position";

export class Circle extends ClosedShape{
    constructor(id:number, size:CircleSize, strokeWidth?:number, 
        enabled?:boolean, strokeColor?:string, 
        fillColor?:string, shapePosition?: Position){

            super( id, size, strokeWidth, 
                enabled, strokeColor, fillColor, shapePosition);

    }

    override setSize(size:CircleSize):void{
        this.size = size;
    }

}